projects/goharbor/harbor/patches/0004-fix-cross-account-issue-for-AWS-ECR.patch

```diff
@@ -0,0 +1,25 @@
+From f9a5af021e8ca00ec5387cc3cbbc2d9786c854f1 Mon Sep 17 00:00:00 2001
+From: Vincent Ni <kangqini@amazon.com>
+Date: Thu, 8 Sep 2022 12:52:32 -0700
+Subject: [PATCH] fix cross account issue for aws ecr
+
+---
+ src/pkg/reg/adapter/awsecr/adapter.go | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/src/pkg/reg/adapter/awsecr/adapter.go b/src/pkg/reg/adapter/awsecr/adapter.go
+index 1ad6ea11c..fdc8fae01 100644
+--- a/src/pkg/reg/adapter/awsecr/adapter.go
++++ b/src/pkg/reg/adapter/awsecr/adapter.go
+@@ -49,7 +49,7 @@ func newAdapter(registry *model.Registry) (*adapter, error) {
+ 		return nil, err
+ 	}
+ 	svc, err := getAwsSvc(
+-		region, registry.Credential.AccessKey, registry.Credential.AccessSecret, registry.Insecure, nil)
++		region, registry.Credential.AccessKey, registry.Credential.AccessSecret, registry.Insecure, &registry.URL)
```

Comment:
Isnt there another change needed to also use this registry.URL to pass into registry ids for the getAuthorizationToken function call

Reply:
With that change, [the following lines](https://github.com/goharbor/harbor/blob/main/src/pkg/reg/adapter/awsecr/auth.go#L116-L118) will be executed so the endpoint is explicitly set there. I tested it and it is working now.

Reply:
Nice